compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+bool remove_no_effect_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleReshape *>(node);
+  if (target_node == nullptr)
+    return false;
+
+  std::vector<int32_t> shape_info;
+  auto new_shape = dynamic_cast<luci::CircleConst *>(target_node->shape());
+  if (new_shape == nullptr)
```

Comment:
Do we have to care dynamic `shape`?

I mean, if `shape` is dynamic, is it possible to remove the reshape operator?

Reply:
On `CircleReshape`, shape could be a `CircleOutputDummy` if only one input comes and updated shape could be in options. so, this is checking shape value passed on `CircleConst` or in option.

Reply:
I see. shape can be an argument or an attribute. Thanks.

Reply:
@jinevening umm... sorry for confusing. I checked on `luci/import` and found that current import logic create if there is no `shape tensor`(second input tensor). so, I think I need to revise this logic little bit.

Reply:
For more detail, after from `luci/import`, shape tensor must be filled.(if `second input` exists, it will pass `second input` to shape. if not, it will pass `CircleConst` from option or `CircleOutputDummy`) So, only handle on `CircleConst` case must be handled.